Python/getargs.c

```diff
@@ -1969,8 +1986,8 @@ _parser_init(void *arg)
     parser->next = _Py_atomic_load_ptr(&_PyRuntime.getargs.static_parsers);
     do {
         // compare-exchange updates parser->next on failure
-    } while (_Py_atomic_compare_exchange_ptr(&_PyRuntime.getargs.static_parsers,
-                                             &parser->next, parser));
+    } while (!_Py_atomic_compare_exchange_ptr(&_PyRuntime.getargs.static_parsers,
```

Comment:
It is intentional.  This is the fix for the secondary bug I mentioned above.  Doing the compare-exchange again after it already succeeded meant it would try one more time and set `parser->next` to `parser`.  That meant only the most recent addition to the linked list was actually in the list, but otherwise this bug was benign.  I had to fix this in order to reproduce the error on main.  The bug doesn't exist on 3.12.

Reply:
oh wow. TBQH there's aren't easy APIs to use (and I can't even reason about them without studying the docs carefully). Thanks for explaining.